Lib/_pyio.py

```diff
@@ -551,6 +551,44 @@ def nreadahead():
                 break
         return bytes(res)
 
+    def backreadline(self, size=-1):
+        r"""Read and return a line of bytes from the end of the stream.
+
+        If size is specified and is an integer, at most size bytes will be
+        read and only the last size bytes of the current line will be returned.
+
+        For instance, using 'backreadline(4)' on 'first line\nsecond line'
+        would return 'line' (and not 'enil' as opposed to 'backread(4)').
+
+        The line terminator is always b'\n' for binary files; for text
+        files, the newlines argument to open can be used to select the
+        line terminator(s) to be recognized.
+        """
+        if size is None:
+            size = -1
+        else:
+            try:
+                size_index = size.__index__
+            except AttributeError:
+                raise TypeError(f"{size!r} is not an integer")
+            else:
+                size = size_index()
+        rev_res, count = bytearray(), 0
+        eol = None  # decide whether a line break is included or not
+        while size < 0 or count < size:
+            b = self.backread(1)
+            if not b:
+                break
+            rev_res += b
+            count += 1
+            if b == b"\n":
+                eol = b
+                break
+        if eol is not None:
+            # reverse the characters in the line, except the new line character
+            return bytes(reversed(rev_res[:-1])) + eol
+        return bytes(reversed(rev_res))
```

Comment:
Could you check whether this is faster? Apparently it's the case for `list(reversed(...))`.

```suggestion
            return rev_res[-2::-1] + eol
        return rev_res[::-1]
```

Reply:
I think I'll leave optimizations once I'm done with the rest. Also, this is the Python implementation and I'm not very worried about performances here IMO.

Reply:
Would probably have to be `bytes(rev_res[::-1])` or `bytes(rev_res)[::-1]`. I'm not going to actively look for optimisations in the Python implementation, this just stood out to me.